### ISSUE TITLE
'ProjectGraphError' from `@nx/vite` breaking all targets and built-in commands

### ISSUE THREAD

**Issue by ontoneio** [nrwl/nx#26344]
### Current Behavior

When running any project target OR built-in NX commands like `nx report` after upgrading to `nx@19.1.2`  yields me a similar error about an extraneous import in another unrelated project in the mono-repo. Issue persists after running `nx reset` reinstalling with pnpm.

### Expected Behavior

Expecting targets not to trigger errors about other unrelated projects, and expect commands like `nx report` to not break in relation to nx project graph issues.

### GitHub Repo

https://github.com/ontoneio/nx-examples/tree/bug/nx-vite-issue

### Steps to Reproduce

1.  run `yarn install`
2. run `yarn nx run products:serve` an unrelated target in both package and build system
3. Observe Error output
4. run `yarn nx reset`
5. re-run `yarn nx run products:serve`
6. Observe same error

```bash
yarn nx run products:serve
yarn run v1.22.19
$ nx run products:serve
node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[Failed to process project graph. Run "nx reset" to fix this. Please report the issue if you keep seeing it.
  The "@nx/vite/plugin" plugin threw an error while creating nodes from apps/vite-bug-repro/vite.config.ts:
    Error: Build failed with 2 errors:
    node_modules/esbuild/lib/main.js:1374:27: ERROR: [plugin: externalize-deps] "@unocss/preset-wind" resolved to an ESM file. ESM file cannot be loaded by `require`. See https://vitejs.dev/guide/troubleshooting.html#this-package-is-esm-only for more details.
    node_modules/esbuild/lib/main.js:1374:27: ERROR: [plugin: externalize-deps] "unocss/vite" resolved to an ESM file. ESM file cannot be loaded by `require`. See https://vitejs.dev/guide/troubleshooting.html#this-package-is-esm-only for more details.
        at failureErrorWithLog (/home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:1651:15)
        at /home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:1059:25
        at runOnEndCallbacks (/home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:1486:45)
        at buildResponseToResult (/home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:1057:7)
        at /home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:1086:16
        at responseCallbacks.<computed> (/home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:704:9)
        at handleIncomingPacket (/home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:764:9)
        at Socket.readFromStdout (/home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:680:7)
        at Socket.emit (node:events:517:28)
        at Socket.emit (node:domain:489:12)
        at addChunk (node:internal/streams/readable:335:12)
        at readableAddChunk (node:internal/streams/readable:308:9)
        at Socket.Readable.push (node:internal/streams/readable:245:10)
        at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)] {
  name: 'ProjectGraphError'
}

Node.js v20.14.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Nx Report

```shell
yarn run v1.22.19
$ nx report

 NX   Report complete - copy this into the issue template

Node   : 20.14.0
OS     : linux-x64
yarn   : 1.22.19

nx                 : 19.1.2
@nx/js             : 19.1.2
@nx/jest           : 19.1.2
@nx/linter         : 19.1.2
@nx/eslint         : 19.1.2
@nx/workspace      : 19.1.2
@nx/angular        : 19.1.2
@nx/cypress        : 19.1.2
@nx/devkit         : 19.1.2
@nx/eslint-plugin  : 19.1.2
@nx/playwright     : 19.1.2
@nx/react          : 19.1.2
@nrwl/tao          : 19.1.2
@nx/vite           : 19.1.2
@nx/web            : 19.1.2
@nx/webpack        : 19.1.2
nx-cloud           : 19.0.0
typescript         : 5.4.3
---------------------------------------
Registered Plugins:
@nx/vite/plugin
@nx/eslint/plugin
@nx/cypress/plugin
@nx/jest/plugin
@nx/playwright/plugin
---------------------------------------
Community plugins:
@ngrx/component-store : 17.0.1
@ngrx/effects         : 17.0.1
@ngrx/entity          : 17.0.1
@ngrx/router-store    : 17.0.1
@ngrx/store           : 17.0.1
@ngrx/store-devtools  : 17.0.1

Done in 0.30s.
```

Check reproduction repo for dependency versions.
```


### Failure Logs

```shell
yarn nx run products:serve
yarn run v1.22.19
$ nx run products:serve
node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[Failed to process project graph. Run "nx reset" to fix this. Please report the issue if you keep seeing it.
  The "@nx/vite/plugin" plugin threw an error while creating nodes from apps/vite-bug-repro/vite.config.ts:
    Error: Build failed with 2 errors:
    node_modules/esbuild/lib/main.js:1374:27: ERROR: [plugin: externalize-deps] "@unocss/preset-wind" resolved to an ESM file. ESM file cannot be loaded by `require`. See https://vitejs.dev/guide/troubleshooting.html#this-package-is-esm-only for more details.
    node_modules/esbuild/lib/main.js:1374:27: ERROR: [plugin: externalize-deps] "unocss/vite" resolved to an ESM file. ESM file cannot be loaded by `require`. See https://vitejs.dev/guide/troubleshooting.html#this-package-is-esm-only for more details.
        at failureErrorWithLog (/home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:1651:15)
        at /home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:1059:25
        at runOnEndCallbacks (/home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:1486:45)
        at buildResponseToResult (/home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:1057:7)
        at /home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:1086:16
        at responseCallbacks.<computed> (/home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:704:9)
        at handleIncomingPacket (/home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:764:9)
        at Socket.readFromStdout (/home/vscode/projects/nx-examples/node_modules/esbuild/lib/main.js:680:7)
        at Socket.emit (node:events:517:28)
        at Socket.emit (node:domain:489:12)
        at addChunk (node:internal/streams/readable:335:12)
        at readableAddChunk (node:internal/streams/readable:308:9)
        at Socket.Readable.push (node:internal/streams/readable:245:10)
        at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)] {
  name: 'ProjectGraphError'
}

Node.js v20.14.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Package Manager Version

1.22.19

### Operating System

- [ ] macOS
- [X] Linux
- [X] Windows
- [X] Other (Please specify)

### Additional Information

I am running on Windows WSL2 Ubuntu Linux 22.04 LTS

**Comment by ontoneio** [comment]
I realize this could be specific to the `UnoCSS from 'unocss/vite' imports configuration for exports, but what I am not understanding is why unrelated projects break with this and why I can't run things like `nx report` because of a simple import error in an unrelated project.  I also kept seeing `Failed to process project graph. Run "nx reset" to fix this. Please report the issue if you keep seeing it.` so I wanted to track down the issue. 

My vite config us using:
```javascript
    commonjsOptions: {
      transformMixedEsModules: true,
    },
```
So I am not sure why such a small thing could break other projects.

**Comment by FrozenPandaz** [comment]
The short explanation is that a plugin failing to construct nodes means that Nx cannot properly construct a task graph for running any tasks. The plugin could have constructed a node which should have been depended upon by the task being run.

The long explanation is... 
* To run tasks, Nx creates a Task Graph.
* The Task Graph is a function of the Project Graph. The definition of projects, and the relationships between them are used to create a task graph with relationships between tasks.
* Nx uses plugins to construct the project graph based on config files in the repo.

So to put it all together in a concrete example...

A serve task you are running could be configured as such meaning that it depends on the build of any of it's dependencies:
```
{
  dependsOn: ['^build']
  ...
}
```

Now, we know.. there's no way that the `jest` plugin which only ever defines `test` targets is going to change this execution... but nothing specifies this to Nx.

So if the `jest` plugin fails to create nodes.. it could have failed to do any number of things it intended to do.
1. It failed to create a test target totally unrelated to the serve task above (this is the reality)
2. It failed to create a build target which would be related to the serve task above
3. It creates a different serve target for that same project which would mean.. the target which should be run should be completely different if the plugin does not fail. 

Ultimately, the Project Graph was not constructed properly.. and Nx is not sure what is missing and whether or not it is important to running the tasks. So Nx will error before running any task.

Does this make sense and answer your question as to why failing to parse a `vite.config.js` is a catastrophic failure?

**Comment by ontoneio** [comment]
@Coly010 What more information do you need? My team and I are still hitting weird errors with out Vite config failing other unrelated projects without a clear reason. Is there anything additional I can add to repro that might be helpful?

**Comment by ontoneio** [comment]
Closing as it seems to be fixed in the latest patches, `19.2.3` thanks everyone.
